akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -572,6 +572,11 @@ trait FlowOps[+Out, +Mat] {
   def grouped(n: Int): Repr[immutable.Seq[Out], Mat] = andThen(Grouped(n))
 
   /**
+   * TODO: description
+   */
+  def limit[T](n: Int): Repr[Out, Mat] = andThen(Limit(n))
```

Comment:
Yes, I think you are correct, we should make it take a `Long`

Originally, the reason was because `limit` is meant to be used with `toSeq` which is only indexable until `Integer.MAX_VALUE` anyway. But I guess people will simply need to `limit(Int.MAX)` if they really want to drain a stream using `toSeq`.